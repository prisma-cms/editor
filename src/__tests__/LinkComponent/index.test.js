import React from 'react';
import { render } from '@testing-library/react'

import linkStateData from "../../dev/Renderer/pages/MainPage/TestApp/mock/data/link";

// const contentStr = `
//   <div>
//     Test content
//     <ul>
//       <li>Row 1</li>
//       <li>Row 2</li>
//     </ul>
//   <div>  
// `;


describe('LinkComponent', () => {

  it('Render Editor with custom LinkComponent', () => {

    /**
     * Mock generateRandomKey();
     * But for production this not allowed, so if you use React.hydrate and SSR,
     * you'l get error for wrong data-offset-key
     */
    jest.mock('draft-js-android-fix/lib/generateRandomKey', () => {

      let count = 0;

      return () => {

        count++;

        return count;
      };
    });

    const Component = require('../../App').default;

    const editorKey = "test-editor";

    const tree = render(<Component
      editorKey={editorKey}
      value={linkStateData}
      LinkComponent={({
        to: href,
        style,
        children,
        ...other
      }) => {
        // console.log("LinkComponent props", other);
        return <a {...other} href={href} style={{ ...style, color: "green" }} className="custom-link">{children}</a>;
      }}
    />)

    expect(tree.container).toMatchSnapshot()

    const node = tree.container.children[0]

    /**
     * Check editorKey
     */
    expect(node.querySelector("[data-editor]").attributes.getNamedItem("data-editor").value).toBe(editorKey);
    
    const customLinks = node.querySelectorAll(".custom-link");
    const allLinks = node.querySelectorAll("a");
    
    /**
     * Check links exists
     */
    expect(customLinks.length).toBeGreaterThan(0);
    
    /**
     * Check all links are custom
     */

    expect(customLinks.length).toEqual(allLinks.length);

  });

})
