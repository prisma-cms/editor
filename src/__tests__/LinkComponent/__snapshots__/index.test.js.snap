// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkComponent Render Editor with custom LinkComponent 1`] = `
<div>
  <div
    class="Component-root-1 "
  >
    <div
      class="DraftEditor-root"
    >
      <div
        class="DraftEditor-editorContainer"
      >
        <div
          aria-describedby="placeholder-test-editor"
          class="public-DraftEditor-content"
          contenteditable="false"
          spellcheck="true"
          style="outline: none; user-select: text; white-space: pre-wrap; word-wrap: break-word;"
        >
          <div
            data-contents="true"
          >
            <div
              class=""
              data-block="true"
              data-editor="test-editor"
              data-offset-key="bhhca-0-0"
            >
              <div
                class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                data-offset-key="bhhca-0-0"
              >
                <span
                  data-offset-key="bhhca-0-0"
                >
                  <span
                    data-text="true"
                  >
                    Link to 
                  </span>
                </span>
                <a
                  class="custom-link"
                  href="https://modxclub.ru"
                  style="color: green;"
                  target="_blank"
                >
                  <span
                    data-offset-key="bhhca-1-0"
                  >
                    <span
                      data-text="true"
                    >
                      modxclub.ru
                    </span>
                  </span>
                </a>
                <span
                  data-offset-key="bhhca-2-0"
                >
                  <span
                    data-text="true"
                  >
                     
                  </span>
                </span>
              </div>
            </div>
            <div
              class=""
              data-block="true"
              data-editor="test-editor"
              data-offset-key="bhhcb-0-0"
            >
              <div
                class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                data-offset-key="bhhcb-0-0"
              >
                <span
                  data-offset-key="bhhcb-0-0"
                >
                  <span
                    data-text="true"
                  >
                    Link to 
                  </span>
                </span>
                <a
                  class="custom-link"
                  href="/content-state"
                  style="color: green;"
                >
                  <span
                    data-offset-key="bhhcb-1-0"
                  >
                    <span
                      data-text="true"
                    >
                      /content-state
                    </span>
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
